libnd4j/include/ops/declarable/helpers/cpu/transforms.cpp

```diff
@@ -758,6 +758,34 @@ void mirrorPad(const NDArray<T>& input, const NDArray<T>& paddings, NDArray<T>&
     }
 }
 
+//////////////////////////////////////////////////////////////////////////
+template<typename T>
+void concat(const std::vector<NDArray<T>*>& inArrs, NDArray<T>& output, const int axis) {
+
+    const int numOfArrs = inArrs.size();
+    const int rank  = inArrs[0]->rankOf();
+    const int rank2 = 2*rank;
+    int* indices = new int[2 * rank * numOfArrs];
+    memset(indices, 0, 2 * rank * numOfArrs * sizeof(int));
+
+    // take into account indices for first array
+    indices[2 * axis + 1] = inArrs[0]->sizeAt(axis);
+    
+    // loop through the rest of input arrays
+    for(int i = 1; i < numOfArrs; ++i) {
+        indices[i * rank2 + 2 * axis]     = indices[2 * axis + 1 + (i-1) * rank2];                                // index start from
+        indices[i * rank2 + 2 * axis + 1] = indices[2 * axis + 1 + (i-1) * rank2] + inArrs[i]->sizeAt(axis);      // index end with (excluding)
+    }    
+
+// #pragma omp parallel for if(numOfArrs > Environment::getInstance()->elementwiseThreshold()) schedule(guided)    
+#pragma omp parallel for schedule(guided)    
+    for(int i = 0; i < numOfArrs; ++i) 
+        output((indices + i * rank2), true).assign(inArrs[i]);
```

Comment:
If this will work fine for all possible cases from java, it will mean that we did some really good work with NDArray.

Reply:
Frankly speaking I'm not sure it will pass all java tests,
anyway we will see...